lib/internal/streams/from.js

```diff
@@ -34,21 +34,49 @@ function from(Readable, iterable, opts) {
     objectMode: true,
     ...opts
   });
+
   // Reading boolean to protect against _read
   // being called before last iteration completion.
   let reading = false;
+
+  // needToClose boolean if iterator needs to be explicitly closed
+  let needToClose = false;
+
   readable._read = function() {
     if (!reading) {
       reading = true;
       next();
     }
   };
+
+  readable._destroy = function(error, cb) {
+    close().then(
+      () => cb(error),
+      (e) => cb(error || e),
```

Comment:
```suggestion
      (e) => process.nextTick(cb, error || e),
```

Reply:
@mcollina which try-catch do you mean?
It seems taht in cb there is no try-catch https://github.com/nodejs/node/blob/b772927daa936df3b2b1ac88df45a5c26204d966/lib/internal/streams/destroy.js#L38

Reply:
Ah.. may be you mean that this call https://github.com/nodejs/node/blob/b772927daa936df3b2b1ac88df45a5c26204d966/lib/internal/streams/destroy.js#L56 can probably has a try-catch?

Reply:
> is this to avoid the unhandledRejection flow?

This makes sense. Now I understand your initial concern about coding style

Reply:
@vadzim: I think @mcollina means the implicit try/catch inside the `Promise.then` implementation which propagates to `unhandledRejection`.

Reply:
fixed

Reply:
@ronag I've got @mcollina initial convert about mixing coding style. I've pushed the fix.

@mcollina Thank you. Very useful catch for me.